decoder/src/elf2table/mod.rs

```diff
@@ -218,10 +218,10 @@ pub fn parse_impl(elf: &[u8], check_version: bool) -> Result<Option<Table>, anyh
 
 /// Checks if the version encoded in the symbol table is compatible with this version of the `decoder` crate
 fn check_version(version: &str) -> Result<(), String> {
-    if version != DEFMT_VERSION {
+    if !DEFMT_VERSIONS.contains(&version) {
         let msg = format!(
             "defmt wire format version mismatch: firmware is using {}, `probe-run` supports {}\nsuggestion: use a newer version of `defmt` or `cargo install` a different version of `probe-run` that supports defmt {}",
```

Comment:
Just a side note: This function is not only used by `probe-run` but also by other binaries. That can become a little bit confusing if `cargo embed` calls itself `probe-run`: https://github.com/rp-rs/rp2040-project-template/issues/57#issuecomment-1530889493
Any idea for a better wording?

Reply:
yeah, this is a bit annoying. we are tracking it here: https://github.com/knurling-rs/defmt/issues/412

While it is ugly, we could just add a second parameter which is the application name.
```rs
fn check_version(version: &str, app_name: &str) -> Result<(), String>
```
```suggestion
            "defmt wire format version mismatch: firmware is using {}, `{app_name}` supports {}\nsuggestion: use a newer version of `defmt` or use a different version of `{app_name}` that supports defmt {}",
```